File: string.py
string = "[Programming [langu[age]]] is the most popular programming language [today]"
substring = "Javascript"


def findClose(string):
    count=1
    for i in range(len(string)):
        if string[i] =='[':
            count +=1
        elif string[i] == ']':
            count -=1
        if string[i]== ']' and count == 1:
            return i

def swap(string, substring):
    start = string.find('[')
    finish = findClose(string)+1
    print(start, finish)
    return substring+string[finish:]

print(swap(string, substring))

File: string.js
let string = "[Programming [langu[age]]] is the most popular programming language [today]"
let substring = "Javascript"


let findClose = (string) => {
    let count = 1
    for (let i=0; i<string.length; i++) {
    
        if (string[i] =='[') {
            count++}
        else if (string[i] == ']') {
            count--}
        if ((string[i]== ']') && (count == 1)) {
            return i
            }
        }
    }

let swap = (string, substring) => {    
    let finish = findClose(string)+1    
    return substring+string.substring(finish)
    }

console.log(swap(string, substring))
